### PR TITLE
feat: add generate example preset for JS

### LINKED ISSUE #625: [📑 Docs]:  Update JavaScript.md with un/marshal function
### What Dev Docs changes are you proposing?

**Description**
To let the user know that `un/marshal feature` of `javascript` is now available to use.

- [ ] Upate `docs/languages/JavaScript.md`.
- [ ] Add examples for this feature in example folder.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Samridhi-98**
**Description**

-  Added `example` option in `javascript/commonpreset.ts`
- Modified `docs/language/javascript`
- Added test for `example` option

**Related issue(s)**

Fixes #625

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@jonaslagoni  please review my PR 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Do you mind splitting this into two PR's? Cause if we ever have to rollback the example feature, we would need to rollback the marshalling example as well, which is not preferred 😄

**Comment by Samridhi-98** [comment]
@jonaslagoni  Sure, will do it.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Hahaha, I had to review make sure I did not ask you to remove the example for the feature just to ask you to add it again 😆 

Remember to create `example` as well as add the documentation for https://github.com/asyncapi/modelina/blob/master/docs/languages/JavaScript.md 🙂

**Comment by Samridhi-98** [comment]
> Hahaha, I had to review make sure I did not ask you to remove the example for the feature just to ask you to add it again 😆
> 
> Remember to create `example` as well as add the documentation for https://github.com/asyncapi/modelina/blob/master/docs/languages/JavaScript.md 🙂

Sorry, I misunderstood it.
I have opened another PR for `examples`  🙂

**Comment by jonaslagoni** [comment]
> Sorry, I misunderstood it. I have opened another PR for `examples` 🙂

You did not misunderstand anything 🙂 The example you have added in your PR, is the one about marshalling right? 

In this PR, we need an example for the `example` preset you have added 🙂

**Comment by Samridhi-98** [comment]
> 

Oh, yeah mistakenly I had added a marshaling example in this PR. 😅

File: test/generators/javascript/preset/utils/ExampleFunctions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { JavaScriptGenerator } from '../../../../../src/generators';
+import { renderValueFromModel } from '../../../../../src/generators/javascript/presets/utils/ExampleFunction';
+import { JavaScriptRenderer } from '../../../../../src/generators/javascript/JavaScriptRenderer';
+import { CommonInputModel, CommonModel } from '../../../../../src/models';
+class MockJavaScriptRenderer extends JavaScriptRenderer {
+}
+const renderer = new MockJavaScriptRenderer(JavaScriptGenerator.defaultOptions, new JavaScriptGenerator(), [], new CommonModel(), new CommonInputModel());
+describe('Marshalling preset', () => {
```
Comment (jonaslagoni) [inline test/generators/javascript/preset/utils/ExampleFunctions.spec.ts:8]: Did you mean something else here? 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Only one small change 💪

File: test/generators/javascript/preset/utils/ExampleFunctions.spec.ts (hunk 1 above)
Comment (Samridhi-98) [reply test/generators/javascript/preset/utils/ExampleFunctions.spec.ts:8]: Sorry, it's a typo `Example preset` 😅

**Review by jonaslagoni — APPROVED** [review]
LGTM 🎉

**Pull request merged**